project/src/main/java/com/google/sps/store/SearchStore.java

```diff
@@ -59,27 +66,22 @@ private void addKeywordsInTextToIndex(String eventId, String text, float weight)
 
     for (Keyword keyword : keywords) {
       float keywordRank = 0;
-      Optional<EventResult> existingResult =
-          keywordToEventResults.get(keyword.getName()).stream()
-              .filter(result -> result.getEventId().equals(eventId))
-              .findFirst();
-      if (existingResult.isPresent()) {
-        keywordRank += existingResult.get().getRanking();
-        keywordToEventResults.remove(keyword.getName(), existingResult.get());
+      Float existingRanking = keywordToRanking.get(keyword.getName());
+      if (existingRanking != null) {
+        keywordRank += existingRanking;
       }
-      keywordRank += weight * keyword.getRelevance();
-      keywordToEventResults.put(keyword.getName(), new EventResult(eventId, keywordRank));
+      keywordRank = Float.sum(keywordRank, weight * keyword.getRelevance());
```

Comment:
nit: Put this in a calculatedWeightedRank type of function? Might help with readability and also if this procedure takes multiple lines in the future

Reply:
I removed the use of Float.sum so it is very small amount of code. Will keep in mind for future but no plans to change algorithm for now.